pkgs/misc/emulators/citra/default.nix

```diff
@@ -1,91 +1,28 @@
-{ lib
-, stdenv
-, fetchFromGitHub
-, cmake
-, wrapQtAppsHook
-, boost17x
-, pkg-config
-, libusb1
-, zstd
-, libressl
-, enableSdl2 ? true, SDL2
-, enableQt ? true, qtbase, qtmultimedia
-, enableQtTranslation ? enableQt, qttools
-, enableWebService ? true
-, enableCubeb ? true, libpulseaudio
-, enableFfmpegAudioDecoder ? true
-, enableFfmpegVideoDumper ? true
-, ffmpeg
-, useDiscordRichPresence ? true, rapidjson
-, enableFdk ? false, fdk_aac
-}:
-assert lib.assertMsg (!enableFfmpegAudioDecoder || !enableFdk) "Can't enable both enableFfmpegAudioDecoder and enableFdk";
-
-stdenv.mkDerivation {
-  pname = "citra";
-  version = "2021-11-01";
-
-  src = fetchFromGitHub {
-    owner = "citra-emu";
-    repo = "citra";
-    rev = "5a7d80172dd115ad9bc6e8e85cee6ed9511c48d0";
-    sha256 = "sha256-vy2JMizBsnRK9NBEZ1dxT7fP/HFhOZSsC+5P+Dzi27s=";
-    fetchSubmodules = true;
+{ branch ? "nightly", libsForQt5, fetchFromGitHub }:
+let inherit libsForQt5 fetchFromGitHub;
+in {
+  nightly = libsForQt5.callPackage ./base.nix rec {
+    pname = "citra-${branch}";
+    version = "1737";
+    branchName = branch;
+    branchDesc = "A Nintendo 3DS Emulator, this branch contains already reviewed and tested features";
+    src = fetchFromGitHub {
```

Comment:
This is one point where split submodules will sure be problematic. Most likely different tags and branches need different versions of their dependencies -- using a single list of them could lead to build failures or crashes.

Reply:
You hit the nail on the head here, this recently became a problem with one dependency being newer in canary than nightly. Of course it's possible to create a temporary workaround but that's an extra burden on maintainers.

Reply:
I'm switching back to fetchSubmodules, thanks for the review!